app/services/atmosphere/cloud/satisfy_appliance.rb

```diff
@@ -39,6 +47,17 @@ def execute
 
                 action.warn(appliance.state_explanation)
               else
+                # Select fund to assign to appliance, if not yet assigned
+                unless appliance.fund.present?
+                  cfs = tenant.funds & appliance.appliance_set.user.funds
+                  if cfs.include? \
+                    appliance.appliance_set.user.default_fund
+                    appliance.fund = appliance.appliance_set.user.default_fund
+                  else
+                    appliance.fund = cfs.first
+                  end
+                end
```

Comment:
Similar as above, extract into separate method and use `||=` to assign fund only when needed.